### ISSUE TITLE
More than one sequential undefined in a sequence gets skipped over

### ISSUE THREAD

**Issue by davemasterdave** [winterbe/sequency#21]
Hi,

The test case below fails because more than one consecutive undefined element in a sequence gets skipped over:

```typescript
const vals =  ["hello", undefined, undefined, undefined, "world"];
expect(asSequence(vals).toArray()).to.deep.equal(vals);
```

The result with typescript 2.6.2 is
```
AssertionError: expected [ 'hello', undefined, 'world' ] to deeply equal [ Array(5) ]
      + expected - actual

       [
         "hello"
         [undefined]
      +  [undefined]
      +  [undefined]
         "world"
       ]
```

Besides this one quirk, this library has been fantastic

**Comment by winterbe** [comment]
Hi.

Thanks for giving Sequency a try. Undefined has a special meaning for many
internal operations. That’s why the result is not as expected.

IMO this edge case should not be a problem in real world projects. If you
replace undefined with null you should get the desired behavior.

Best regards.

Benjamin


davemasterdave <notifications@github.com> schrieb am Di. 2. Jan. 2018 um
02:21:

**Comment by davemasterdave** [comment]
I am working on a large typescript server application and a lot of the interface definitions have optional parameters. Furthermore, the project has strictNullChecks enabled so all of the optional definitions result in types of ```T | undefined```, resulting in making the ```null``` type incompatible.

I ran into this quirk since a lot of the types are possibly also of the type "undefined". More specifically, I had an iterable fixed length data structure that happened to have several optional types that were not defined in a row. Then, when I was using this library to convert the fixed length object into another iterable object of the same length, I noticed that the output was shorter than what it was supposed to be.

Best Regards

David

**Comment by winterbe** [comment]
Thanks for explaining your usecase. I will look into this next week so we
can properly support undefined values in Sequency.


davemasterdave <notifications@github.com> schrieb am Mi. 3. Jan. 2018 um
02:52:

**Comment by winterbe** [comment]
Solving #15 should also fix this.

**Comment by davemasterdave** [comment]
Ok sounds good

**Comment by winterbe** [comment]
@davemasterdave The issue is fixed with next version 0.19.0 which will be published soon. Sorry for the delay
